tools/cli_helper.class.inc

```diff
@@ -0,0 +1,181 @@
+<?php declare(strict_types=1);
+/**
+ *
+ * This class is designed to assist in writing command-line interface tools
+ * for LORIS.
+ *
+ * It contains helper functions to format output, etc.
+ * PHP Version 7
+ *
+ * @category Tools
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+require_once 'generic_includes.php';
+/**
+ * This class is designed to assist in writing command-line interface tools
+ * for LORIS.
+ *
+ * It contains helper functions to format output, etc.
+ *
+ * @category Tools
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class CLI_Helper
+{
+    // Colours associated with the bash shell.
+    // @see https://misc.flogisoft.com/bash/tip_colors_and_formatting
+    const COLOURS = [
+        'red' => '31',
+        'yellow' => '33',
+        'green' => '32',
+    ];
+    private $_loggingEnablded = false;
+    private $_logDirectory;
+
+    /**
+     *
+     */
+    public function __construct(SPLFileInfo $logDirectory = null)
+    {
+        $_logDirectory = new SPLFileInfo(
+            \NDB_Factory::singleton()
+                ->config()
+                ->getSetting('log')
+        );
+    }
+
+    /**
+     * Print a formatted and indented bullet point.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    function printBulletPoint(string $line): void
+    {
+        echo "\t* $line\n";
+    }
+
+    /**
+     * Prints a line of text with a newline before and after. Can also be
```

Comment:
I changed it to print only a line feed at the beginning and updated the comment.

The other print function are wrappers around this one and they handle colours and prefixes. They do log automatically since they pass through this function. So I think your idea is already present in the code.